op-challenger/game/service.go

```diff
@@ -34,10 +36,11 @@ func NewService(ctx context.Context, logger log.Logger, cfg *config.Config) (*Se
 		return nil, fmt.Errorf("failed to create the transaction manager: %w", err)
 	}
 
-	client, err := client.DialEthClientWithTimeout(client.DefaultDialTimeout, logger, cfg.L1EthRpc)
+	rpcClient, err := client.DialRPCClientWithTimeout(client.DefaultDialTimeout, logger, cfg.L1EthRpc)
 	if err != nil {
 		return nil, fmt.Errorf("failed to dial L1: %w", err)
 	}
+	l1Client := ethclient.NewClient(rpcClient)
```

Comment:
I'd avoid this change.  We can just use `DialEthClientWithTimeout` and then use its `.Client()` method to get the underlying RPC client. It's not a big deal, but nice to pass as much of the setup as possible to the util function rather than assembling the pieces here.

Reply:
```suggestion
		l1Client, err := client.DialEthClientWithTimeout(client.DefaultDialTimeout, logger, cfg.L1EthRpc)
```